timemachine/kernels/wrap_kernels.cpp

```diff
@@ -0,0 +1,159 @@
+#include <pybind11/pybind11.h>
+#include <pybind11/stl.h>
+#include <pybind11/numpy.h>
+
+#include "cpu/bonded_kernels.hpp"
+
+namespace py = pybind11;
+
+// hessian matrix product
+template<typename RealType>
+py::tuple declare_harmonic_bond_hmp(
+    const py::array_t<RealType, py::array::c_style> &coords,
+    const py::array_t<RealType, py::array::c_style> &params,
+    const py::array_t<RealType, py::array::c_style> &dxdps,
```

Comment:
@bp-kelley -  I'd like this to be an array or None

Reply:
You can’t really send none to a c++ reference.  You could just have empty arrays with:

 const py::array_t<RealType, py::array::c_style> dxps =  py::array_t<RealType, py::array::c_style> ()

Other wise use a unique_ptr to the optional array.

If 

As a default arg.